### PR TITLE
[WIP] API Documentation

### PR DESCRIPTION
**Pull request by zdavis**

### CODE REVIEW

File: api/spec/swagger_definitions/comments.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            flags_count: Type.integer,
+            flagged: Type.boolean,
+            abilities: Type.object( Type.crud ), # TODO check that crud is the right abilities type
+            deleted: Type.boolean,
+            children_count: Type.integer,
+            sort_order: Type.integer,
+            author_created: Type.boolean,
+          }),
+          relationships: Type.object({
+            creator: Type.reference( '#/definitions/Maker' ) # TODO check that this is indeed a maker type
```
Comment (zdavis) [inline api/spec/swagger_definitions/comments.rb]: The `creator` relationship will always (or almost always) be a User, not a Maker.

File: api/spec/requests/contacts_controller_spec.rb [hunk 2]
```diff
@@ -19,25 +19,31 @@
...
+
+      response '422', I18n.t('swagger.unprocessable') do
+        let(:contact_params) {{ data: invalid_params }}
+        schema '$ref' => '#/definitions/ContactResponseError'
+        run_test!
       end
     end
+  end
 
+  describe "sends the message to installation" do
```
Comment (zdavis) [inline api/spec/requests/contacts_controller_spec.rb]: This language is weird (not your fault, you didn't write it). Let's change it to "creates a new contact email"

File: api/spec/swagger_definitions/content_blocks.rb [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+          }),
+          relationships: Type.object({
+            project: Type.object({
+              data: Type.object({
+                id: Type.id,
+                type: Type.string,
+              })
+            }),
+            featuredCollections: Type.object({
+              data: Type.array( type: Type.string ) #TODO I don't know what this data is
```
Comment (zdavis) [inline api/spec/swagger_definitions/content_blocks.rb]: It's an array of references to ResourceCollection models:

image

File: api/spec/swagger_definitions/ingestions.rb [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    #
+    #   param_config = structure_params(
+    #     attributes: attributes,
+    #     relationships: relationships
+    #   )
+    #
+    #   params.permit(param_config)
+    # end
+
+    def request_create
```
Comment (zdavis) [inline api/spec/swagger_definitions/ingestions.rb]: A sample request for ingestion via upload:
image

A sample request for ingestion via URL:
image

File: api/spec/swagger_definitions/comments.rb (hunk 1 above)
Comment (ConnorChristensen) [reply api/spec/swagger_definitions/comments.rb]: Is that the case as well for the creators on the `project` type?

```
    def request_create
      Type.object({
        data: Type.object({
          attributes: Type.object(request_create_attributes, { required: required_create_attributes }),
          relationships: Type.object({
            creators: Type.reference('#/definitions/Maker'),
            collaborators: Type.reference('#/definitions/Maker'),
            contributors: Type.reference('#/definitions/Maker'),
            subjects: Type.reference('#/definitions/Subject')
          })
        })
      })
    end
```

Comment (zdavis) [reply api/spec/swagger_definitions/comments.rb]: No. I'm referring specifically to the `creator` relationship. I believe that ever time you see a belongs_to `creator` relationship, it is a relationship to a single user object. In the case of the project, which has_many creators, it is a reference to makers.

On projects, creators is an array of makers. contributors is an array of makers, and collaborators is an array of collaborators, which is a different model from maker. See app/models/concerns/collaborative.rb for some more details about what's happening here.

**Comment by zdavis** [comment]
Closing this in favor of the newer request.

**Pull request closed without merging**